README.md

```diff
@@ -1,53 +1,60 @@
+# Scale to Zero
 
+Infrastructure to scale any resource to and from zero
+
+## Prerequisites
+
+**Custom metrics API implementation:**
+
+The Autoscaler takes decisions based on data queried from Kubernetes 
+[custom metrics API.](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/custom-metrics-api.md)
+There are several possible tools that implement it, we internally use
+[Prometheus](https://prometheus.io/) with the [Prometheus-Adapter](https://github.com/DirectXMan12/k8s-prometheus-adapter)
+but you can use which ever you want... you can find [here](https://github.com/kubernetes/metrics/blob/master/IMPLEMENTATIONS.md#custom-metrics-api) 
+recommended implementations
+
+## Design:
+![Design scheme](https://www.lucidchart.com/publicSegments/view/cc8927a6-537f-4fe6-95e1-731503bc7996/image.jpeg)
+
+**The autoscaler** - responsible to periodically check whether resources should be scale to zero, it does that by 
+querying the custom metrics API. When it decide a resource should be scaled to zero it tells the resource scaler to 
+scale the resource to zero.
+The resource scaler will route the traffic to the DLX, which in terms of K8s is done by changing the svc selector, 
+after that, it will scale the resource to zero.
+
+**The DLX** - responsible for receiving and buffering requests of scaled to zero resources, when it gets a message it 
+creates a buffer for the messages, and tells the resource scaler to scale the service back from zero.
+The resource scaler will scale the resource back up and then route the traffic back to the service (by changing the svc 
+selector).
+
+## Getting Started
+This infrastructure designed to be generic and extendable, it can scale any resource.
+All you have to do is implement the resource scaler, the interface between it and this infrastructure's components is 
+defined in [scaler-types](https://github.com/v3io/scaler-types)
+
+**Note:** Incompatibility between the scaler repo vendor dir and the resource scale vendor dir may break things, 
+therefore it's suggested to put the resource scaler in its own repo
+
+Examples:
+* [Nuclio functions resource scaler](https://github.com/nuclio/nuclio/blob/master/pkg/platform/kube/resourcescaler/resourcescaler.go)
+* [Iguazio's app service resource scaler](https://github.com/v3io/app-resource-scaler/blob/development/resourcescaler.go)  
+
+## Installing
+Go plugin is the magic that glues the resource scaler and this infrastructure components together
+First you'll need to build a Dockerfile that builds your resource scaler as a Go plugin, and transplant it in this 
```

Comment:
- Does it have to be a dockerfile? 🤷‍♂ if the user just builds a go binary why is that not enough?
- Transplant is not clear in this context imo. If I read this I wouldn't know what to do without going to the examples. Explain in a few sentences

Reply:
The whole context here is Kubernetes, in the design chart It's written Autoscaler **pod** etc... If it's a pod it's an image, and if it's an image there's a dockerfile...
It is possible to run it as a binary, but that doesn't make much sense.

Reply:
it's all true but a stretch to the reader to understand all that from the word "transplant" :)